src/components/ChannelPreview/ChannelPreviewMessenger.js

```diff
@@ -63,104 +64,68 @@ const Message = styled.Text`
-      return (
-        <Avatar image={channel.data.image} size={40} name={channel.data.name} />
-      );
-
-    if (otherMembers.length === 1)
-      return (
-        <Avatar
-          image={otherMembers[0].user.image}
-          size={40}
-          name={channel.data.name || otherMembers[0].user.name}
-        />
-      );
-
-    return <Avatar size={40} name={channel.data.name} />;
-  };
-
-  render() {
-    const { channel, t } = this.props;
-    let otherMembers = [];
-    let name = channel.data.name;
-    const isValidName = name && typeof name === 'string';
-
-    if (!isValidName) {
-      const members = channel.state ? Object.values(channel.state.members) : [];
-      otherMembers = members.filter(
-        (member) => member.user.id !== this.props.client.userID,
-      );
-      name = otherMembers
-        .map((member) => member.user.name || member.user.id || 'Unnamed User')
-        .join(', ');
-    }
-    const formatLatestMessageDate = this.props.formatLatestMessageDate;
-    return (
-      <Container onPress={this.onSelectChannel}>
-        {this.renderAvatar(otherMembers)}
-        <Details>
-          <DetailsTop>
-            <Title ellipsizeMode="tail" numberOfLines={1}>
-              {name}
-            </Title>
-            <Date>
-              {formatLatestMessageDate
-                ? formatLatestMessageDate(
-                    this.props.latestMessage.messageObject.created_at,
-                  )
-                : this.props.latestMessage.created_at}
-            </Date>
-          </DetailsTop>
-          <Message
-            unread={this.props.unread > 0 ? this.props.unread : undefined}
-          >
-            {!this.props.latestMessage
-              ? t('Nothing yet...')
-              : truncate(this.props.latestMessage.text.replace(/\n/g, ' '), {
-                  length: this.props.latestMessageLength,
-                })}
-          </Message>
-        </Details>
-      </Container>
-    );
-  }
-}
-
-export default withTranslationContext(themed(ChannelPreviewMessenger));
+const ChannelPreviewMessenger = ({
+  channel,
+  latestMessage,
+  unread,
+  latestMessageLength = 30,
+  formatLatestMessageDate,
+  setActiveChannel,
+}) => {
+  const displayName = useChannelPreviewDisplayName(channel);
+  const displayAvatar = useChannelPreviewDisplayAvatar(channel);
+
+  return (
+    <Container
+      onPress={setActiveChannel.bind(null, channel)}
+      testID="channel-preview-button"
+    >
+      <Avatar size={40} image={displayAvatar.image} name={displayAvatar.name} />
+      <Details>
+        <DetailsTop>
+          <Title ellipsizeMode="tail" numberOfLines={1}>
+            {displayName}
+          </Title>
+          <Date>
+            {formatLatestMessageDate
+              ? formatLatestMessageDate(latestMessage.messageObject.created_at)
+              : latestMessage.created_at}
+          </Date>
+        </DetailsTop>
+        <Message unread={unread > 0 ? unread : undefined}>
```

Comment:
what does `unread` do? I just looked through all of Message/MessageSimple and it doesn't seem to be used anywhere

Reply:
this Message is a styled component, so it's used to toggle some styles

Reply:
lol smh styled components